cw_decred/lib/api/spvwallet.dart

```diff
@@ -1,29 +1,39 @@
-import 'package:cw_core/wallet_info.dart';
+import 'dart:convert';
+import 'dart:io';
 import 'package:cw_core/sync_status.dart';
+import 'package:cw_core/transaction_direction.dart';
 import 'package:cw_core/unspent_transaction_output.dart';
+import 'package:cw_decred/api/libdcrwallet.dart';
+import 'package:cw_decred/api/util.dart';
 import 'package:cw_decred/balance.dart';
 import 'package:cw_decred/pending_transaction.dart';
 import 'package:cw_decred/transaction_info.dart';
-import 'package:cw_core/transaction_direction.dart';
-import 'dart:typed_data';
-import 'dart:io';
 
-// Will it work if none of these are async functions?
 class SPVWallet {
-  SPVWallet();
-
-  SPVWallet create(Uint8List seed, String password, WalletInfo walletInfo) {
-    return SPVWallet();
-  }
-
-  SPVWallet load(String password, String name, WalletInfo walletInfo) {
-    return SPVWallet();
+  final String name;
+  // password is currently only used for seed display, but would likely also be
+  // required to sign inputs when creating transactions.
+  final String password;
+
+  SPVWallet({required this.name, required this.password});
+
+  String? get seed {
+    final walletName = name.toCString();
+    final pass = password.toCString();
+    final seed = dcrwalletApi.walletSeed(walletName, pass);
+    freePointers([walletName, pass]);
+    return seed.toDartString();
   }
 
   DecredBalance balance() {
+    final walletName = name.toCString();
+    final balJson = dcrwalletApi.walletBalance(walletName).toDartString();
+    walletName.free();
+
+    final map = jsonDecode(balJson!);
     return DecredBalance(
-      confirmed: 777,
-      unconfirmed: 111,
+      confirmed: map["confirmed"] ?? 0,
+      unconfirmed: map["unconfirmed"] ?? 0,
```

Comment:
doesn't the tx data just contain `confirmations`?

where does confirmed and unconfirmed come from

unconfirmed : confirmations < 2
confirmed : confirmations >= 2

Reply:
i found my answer here https://github.com/itswisdomagain/libwallet/pull/2#discussion_r1396526014